tests/test_aside.py

```diff
@@ -141,48 +159,58 @@ def test_responses(self):
         # replace(deprecated=True) doesn't work for BlockUsageLocator
         problem_id = BlockUsageLocator(course_id, problem_id.block_type, problem_id.block_id, deprecated=True)
 
+        run1 = RapidResponseRun.objects.create(
+            course_key=course_id,
+            problem_usage_key=problem_id,
+            open=False
+        )
+        run2 = RapidResponseRun.objects.create(
+            course_key=course_id,
+            problem_usage_key=problem_id,
+            open=True
+        )
+
         # This problem is imported into the modulestore in the setup method.
         # It needs to be present there to allow the view function to look up problem data.
-        request = RequestFactory().request()
-        request.user = self.instructor
-        request.session = request.environ
         problem = self.get_problem_by_id(problem_id)
         aside_block = get_aside_from_xblock(problem, self.aside_usage_key.aside_type)
 
-        answer_id_text_counts = zip(
-            range(3),
-            [
-                'an incorrect answer',
-                'the correct answer',
-                'a different incorrect answer',
-            ],
-            range(2, 5),
-        )
-        answer_data = [
-            {
-                'answer_id': 'choice_{}'.format(i),
-                'answer_text': text,
-            }
-            for i, text, _ in answer_id_text_counts
+        answers = [
+            ('choice_0', 'an incorrect answer'),
+            ('choice_1', 'the correct answer'),
+            ('choice_2', 'a different incorrect answer'),
         ]
-        counts = {'choice_{}'.format(ans_id): ans_count for ans_id, _, ans_count in answer_id_text_counts}
+        answers_lookup = {answer_id: answer_text for answer_id, answer_text in answers}
+        counts = list(zip(
+            ['choice_{}'.format(i) for i in range(3)],
+            range(2, 5),
+            [run1.id, run1.id, run1.id],
+        )) + list(zip(
+            ['choice_{}'.format(i) for i in range(3)],
+            [3, 0, 7],
+            [run2.id, run2.id, run2.id],
+        ))
 
-        for item in answer_data:
-            for number in range(counts[item['answer_id']]):
-                username = 'user_{}_{}'.format(number, item['answer_id'])
-                email = 'user{}{}@email.com'.format(number, item['answer_id'])
-                user = User.objects.create(
+        counts_dict = defaultdict(dict)
+        for answer_id, count, run_id in counts:
+            counts_dict[answer_id][str(run_id)] = count
```

Comment:
This test case is pretty long (almost 100 lines) and this block (up to line 183) is adding a lot of cognitive load. If you can think of a straightforward way to break this up or make the expectations more explicit, that would be helpful. Might be a good option to break out some of the `responses` handler logic into helper methods, test those methods, then patch them as part of this test case

Reply:
I had the same concern, I'll see what I can do